pyam/iiasa.py

```diff
@@ -589,31 +590,42 @@ def _new_default_api(kwargs):
 def read_iiasa(
     name, default_only=True, meta=True, creds=None, base_url=_AUTH_URL, **kwargs
 ):
-    """Query an IIASA Scenario Explorer database API and return as IamDataFrame
+    """Read data from an |ixmp4| platform or an IIASA Scenario Explorer database.
 
     Parameters
     ----------
     name : str
-        | Name of an IIASA Scenario Explorer database instance.
+        | Name of an |ixmp4| platform or an IIASA Scenario Explorer database instance.
+        | Use :attr:`platforms <pyam.iiasa.platforms>` for a list of |ixmp4| platforms
+          hosted by IIASA.
         | Use :attr:`valid_connections <pyam.iiasa.Connection.valid_connections>`
-          for a list of available instances.
+          for a list of available Scenario Explorer database instances.
     default_only : bool, optional
         If `True`, return *only* the default version of a model/scenario.
         If `False`, return all versions.
     meta : bool or list of strings, optional
         If `True`, include all meta categories & quantitative indicators
         (or subset if list is given).
     creds : str or :class:`pathlib.Path`, optional
-        | Credentials (username & password) are not required to access
-          any public Scenario Explorer instances (i.e., with Guest login).
-        | See :class:`pyam.iiasa.Connection` for details.
-        | Use :meth:`pyam.iiasa.set_config` to set credentials
-          for accessing private/restricted Scenario Explorer instances.
+        Path to a file with authentication credentials. This feature is deprecated,
+        please run ``ixmp4 login <username>`` in a console instead.
     base_url : str
-        Authentication server URL
-    kwargs
-        Arguments for :meth:`pyam.iiasa.Connection.query`
+        Authentication server URL.
+    **kwargs
+        Arguments for :meth:`pyam.read_ixmp4` or :meth:`pyam.iiasa.Connection.query`.
+
+    Notes
+    -----
+    Credentials (username & password) are not required to access any public |ixmp4|
+    or Scenario Explorer database (i.e., with Guest login).
     """
+    if name in [i.name for i in ixmp4.conf.settings.manager.list_platforms()]:
+        if meta is not True:
+            raise NotImplementedError(
```

Comment:
If there are name overlaps between a Scenario Explorer and an ixmp4 platform I think we could run into an issue here. 
If a user wants `meta=True` but with a Scenario Explorer we could wrongly get an error here.